testpb/1.pulsar.go

```diff
@@ -976,7 +1007,1122 @@ func (x *fastReflection_A) IsValid() bool {
+			n += 2
+		}
+		if x.INT32 != 0 {
+			n += 1 + runtime.Sov(uint64(x.INT32))
+		}
+		if x.SINT32 != 0 {
+			n += 1 + runtime.Soz(uint64(x.SINT32))
+		}
+		if x.UINT32 != 0 {
+			n += 1 + runtime.Sov(uint64(x.UINT32))
+		}
+		if x.INT64 != 0 {
+			n += 1 + runtime.Sov(uint64(x.INT64))
+		}
+		if x.SING64 != 0 {
+			n += 1 + runtime.Soz(uint64(x.SING64))
+		}
+		if x.UINT64 != 0 {
+			n += 1 + runtime.Sov(uint64(x.UINT64))
+		}
+		if x.SFIXED32 != 0 {
+			n += 5
+		}
+		if x.FIXED32 != 0 {
+			n += 5
+		}
+		if x.FLOAT != 0 || math.Signbit(float64(x.FLOAT)) {
+			n += 5
+		}
+		if x.SFIXED64 != 0 {
+			n += 9
+		}
+		if x.FIXED64 != 0 {
+			n += 9
+		}
+		if x.DOUBLE != 0 || math.Signbit(x.DOUBLE) {
+			n += 9
+		}
+		l = len(x.STRING)
+		if l > 0 {
+			n += 1 + l + runtime.Sov(uint64(l))
+		}
+		l = len(x.BYTES)
+		if l > 0 {
+			n += 2 + l + runtime.Sov(uint64(l))
+		}
+		if x.MESSAGE != nil {
+			l = proto.Size(x.MESSAGE)
+			n += 2 + l + runtime.Sov(uint64(l))
+		}
+		if len(x.MAP) > 0 {
+			sortme := make([]string, 0, len(x.MAP))
+			for k := range x.MAP {
+				sortme = append(sortme, k)
+			}
+			sort.Strings(sortme)
+			for _, k := range sortme {
+				v := x.MAP[k]
+				_ = k
+				_ = v
+				l = 0
+				if v != nil {
+					l = proto.Size(v)
+				}
+				l += 1 + runtime.Sov(uint64(l))
+				mapEntrySize := 1 + len(k) + runtime.Sov(uint64(len(k))) + l
+				n += mapEntrySize + 2 + runtime.Sov(uint64(mapEntrySize))
+			}
+		}
+		if len(x.LIST) > 0 {
+			for _, e := range x.LIST {
+				l = proto.Size(e)
+				n += 2 + l + runtime.Sov(uint64(l))
+			}
+		}
+		switch x := x.ONEOF.(type) {
+		case *A_ONEOF_B:
+			if x == nil {
+				break
+			}
+			if x.ONEOF_B != nil {
+				l = proto.Size(x.ONEOF_B)
+				n += 2 + l + runtime.Sov(uint64(l))
+			}
+		case *A_ONEOF_STRING:
+			if x == nil {
+				break
+			}
+			l = len(x.ONEOF_STRING)
+			if l > 0 {
+				n += 2 + l + runtime.Sov(uint64(l))
+			}
+		}
+		if len(x.LIST_ENUM) > 0 {
+			l = 0
+			for _, e := range x.LIST_ENUM {
+				l += runtime.Sov(uint64(e))
+			}
+			n += 2 + runtime.Sov(uint64(l)) + l
+		}
+		if x.Imported != nil {
+			l = proto.Size(x.Imported)
+			n += 2 + l + runtime.Sov(uint64(l))
+		}
+		if x.unknownFields != nil {
+			n += len(x.unknownFields)
+		}
+		return protoiface.SizeOutput{
+			NoUnkeyedLiterals: struct{}{},
+			Size:              n,
+		}
+	}
+
+	marshal := func(input protoiface.MarshalInput) (protoiface.MarshalOutput, error) {
+		x := input.Message.Interface().(*A)
+		if x == nil {
+			return protoiface.MarshalOutput{
+				NoUnkeyedLiterals: struct{}{},
+				Buf:               input.Buf,
+			}, nil
+		}
+		size := proto.Size(x)
+		dAtA := make([]byte, size)
+		i := len(dAtA)
+		_ = i
+		var l int
+		_ = l
+		if x.unknownFields != nil {
+			i -= len(x.unknownFields)
+			copy(dAtA[i:], x.unknownFields)
+		}
+		switch x := x.ONEOF.(type) {
+		case *A_ONEOF_B:
+			if x.ONEOF_B != nil {
+				encoded, err := proto.MarshalOptions{Deterministic: true}.Marshal(x.ONEOF_B)
```

Comment:
Here we should propagate the UseCachedSize flag too which comes from MarshalInputFlags